logos/src/logos/pipeline/base_scheduler.py

```
@@ -294,6 +294,20 @@ def reevaluate_model_queues(self, model_name: str) -> None:
+            # `ClassificationCorrectingScheduler._compute_candidate_scores`:
+            # `_model_registry` is DB-derived and includes every deployment
+            # regardless of session state. Without this check, a queued
+            # future could be dispatched onto a worker whose session was
+            # popped after enqueue — the pipeline would then crash at
+            # execution-context resolution with
+            # LogosNodeOfflineError("No active logosnode worker session").
+            try:
+                if not self._logosnode.is_provider_online(provider_id):
+                    continue
+            except Exception:
+                continue
```

Ruff (0.15.15)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)